### PR TITLE
null safety 

### PR DESCRIPTION
**Pull request by fzyzcjy**
https://github.com/theyakka/qr.flutter/issues/127

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -6,18 +6,18 @@ version: 3.2.0
 homepage: https://github.com/lukef/qr.flutter
```
Comment (vishnuagbly) [inline pubspec.yaml:6]: you are supposed to upgrade the version number here too, either to 4.0.0
also update CHANGELOG.md

**Review by vishnuagbly — CHANGES_REQUESTED** [review]
pubspec.yaml and CHANGELOG.md is needed to be updated.

**Comment by fzyzcjy** [comment]
@vishnuagbly done

**Review by vishnuagbly — APPROVED** [review]

**Comment by fzyzcjy** [comment]
@vishnuagbly Hi I wonder what should I do now? Seems that the PR has not been merged yet

**Comment by vishnuagbly** [comment]
@lukef can you review and merge this pull request.

File: lib/src/qr_image.dart [hunk 2]
```diff
@@ -233,10 +233,10 @@ class _QrImageState extends State<QrImage> {
   }
 
   Widget _errorWidget(
-      BuildContext context, BoxConstraints constraints, Object error) {
+      BuildContext context, BoxConstraints constraints, Object? error) {
     final errorWidget = widget.errorStateBuilder == null
         ? Container()
-        : widget.errorStateBuilder(context, error) ?? Container();
+        : widget.errorStateBuilder!(context, error) ?? Container();
```
Comment (koenusz) [inline lib/src/qr_image.dart:239]: I used this branch in my app to test it a bit. My app seemed to work fine but duing assembly flutter was yelling at me:
```
../../snap/flutter/common/flutter/.pub-cache/git/qr.flutter-2648de0844ec9a2b9e463edc527d29362506105d/lib/src/qr_image.dart:239:36: Warning: Operand of null-aware operation '??' has type 'Widget' which excludes null.
 - 'Widget' is from 'package:flutter/src/widgets/framework.dart' ('../../snap/flutter/common/flutter/packages/flutter/lib/src/widgets/framework.dart').
        : widget.errorStateBuilder!(context, error) ?? Container();
```

The return type of   QrErrorBuilder is Widget so the ?? Container() is not necessary here.

Comment (fzyzcjy) [reply lib/src/qr_image.dart]: fixed

**Comment by jonas-zebari** [comment]
Shall I close #128 in favor of this PR?

**Comment by fzyzcjy** [comment]
@jonas-zebari As long as the null safety feature can be released soon ;)

**Comment by vishnuagbly** [comment]
@lukef can you review this request or maybe add some more maintainers to this repo, I would be happy to volunteer.

**Comment by lukepighetti** [comment]
closed by https://github.com/theyakka/qr.flutter/pull/130

**Pull request closed without merging**